server/src/service.rs

```diff
@@ -123,21 +125,36 @@ impl Service {
 }
 
 impl Service {
-    pub async fn run() -> Result<()> {
+    pub async fn new(secret: Option<Secret>) -> Self {
+        if secret.is_none() {
+            // todo: may be create a temporary Keyring here
+            panic!("Keyring secret can't be empty");
```

Comment:
The service should not really care about the keyring secret, you care about it when starting the login collection, it should be handled there, in a separate method

Reply:
Yeah, implemented a new method.